lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -4101,6 +4106,22 @@ type RewardAccountBuilder k
         =  (k 'RootK XPrv, Passphrase "encryption")
         -> (XPrv, Passphrase "encryption")
 
+withRecentEra
+    :: AnyCardanoEra
+    -> (forall e. WriteTx.IsRecentEra e => WriteTx.RecentEra e -> Handler a)
+    -> Handler a
+withRecentEra anyCardanoEra handleRecentEra = do
+    let invalidEra = ErrMkTransactionInvalidEra anyCardanoEra
```

Comment:
```haskell
let invalidEra = ErrMkTransactionInvalidEra anyCardanoEra
```
looks wrong, as it renders as
```haskell
        ErrMkTransactionInvalidEra _era ->
            apiError err500 CreatedInvalidTransaction $ mconcat
                [ "Whoops, it seems like I just experienced a hard-fork in the "
                , "middle of other tasks. This is a pretty rare situation but "
                , "as a result, I must throw away what I was doing. Please "
                , "retry your request."
                ]
```

I think we want something like `ErrOldEraNotSupported` with the following rendering
```
        ErrOldEraNotSupported (Cardano.AnyCardanoEra era) ->
            apiError err403 BalanceTxEraNotSupported $ T.unwords
                [ "Balancing in ", showT era, " "
                , "is not supported."
                ]
```

_Ideally_ it would say something else than `Balancing in` when called from other endpoints, like constructTx, or quitStakePool. Maybe it could be wrapped somehow. Or maybe we just go with `ErrOldEraNotSupported` 🤔

Reply:
"Creating transactions in old <era> is not supported"? "Constructing"?

Sticking with `ErrOldEraNotSupported` for simplicity sounds sane at least.